### ISSUE TITLE
Outdated CHANGELOG.md file

### ISSUE THREAD

**Issue by mnasyrov** [techfort/LokiJS#575]
CHANGELOG.md file is stopped on 1.3 version while new versions were released (current is 1.5). Also project's releases page has no useful info as well.
Without a project history it is hard to follow up new versions.

**Comment by obeliskos** [comment]
I will look into retroactively piecing together a few of the recent releases in the next week.

The most recent 1.5 release bumped a minor version mainly because of changes to binary indices.  Binary indices before 1.5 were susceptible to corruption if values of mixed types or types that serialized to different types were involved.   Binary indices were overhauled to be 'loosely' sorted in custom groups where : 
- Nan/null/undefined are always less than 
- Number-like values such as 11, "111", 1.53  (and now dates)
- Dates (when binary indexed) are converted to epoch dates (numbers) to avoid corruption of index when serialized... dates would not have worked with binary indices before and would only have made the index 'out-of-order' after serializing to string.  There is a new ['serializableIndices'](https://rawgit.com/techfort/LokiJS/master/jsdoc/Collection.html) collection constructor option where you can disable but as previously mentioned that index would be ineffective on dates in js date object format.
- String, Arrays and objects occur after that

So range ops (on mixed types with binary index) on strings starting with numbers will be different... for instance '5' < '22', and '99' < '1a'.  If you are really relying on string sorting with leading numbers you may want to not use binary index on that column or disable it by performing an op in a chain such as a sort (since only the first op would be able to use index).

In all, more resilient indices with dirty data will be necessary as many times users can not monitor every object they import to analyze for dirty data which may prove problematic.

We also automatically rebuild the indices on db load for any database saved prior to 1.5 upgrade.  Large databases might notice a one-time startup delay.

Other minor features we added : 
- implicit $and syntax : 
```javascript
coll.find({ a:1, b:2 });
```
- fixed LokiFsStructuredAdapter error when first time run on database and it does not exist yet.
- fixed simplesort descending when property is indexed and no filters have been applied
- fix for nativescript environment
- collection 'insert' events no longer emit with internal object reference when 'clone' option is set
- fix to clone method 'shallow'
- added 'removeMeta' option to chained data() calls which will shallow clone and remove $loki and meta properties
- fixes to async unit tests 
- npm release trimmed from 5megs to about 500k by stripping out unnecessary items, git clone if you want examples or jsdocs
- added quickstart examples (for git cloners)
